tracer/src/Datadog.Trace/Util/RequestDataHelper.cs

```diff
@@ -77,7 +77,7 @@ internal static QueryString GetQueryString(HttpRequest request)
         {
             return request.Form;
         }
-        catch (HttpRequestValidationException)
+        catch (HttpException)
```

Comment:
At this point, I have to wonder why we're not just catching _all_ exceptions here 🤔 What exceptions would we _want_ to bubble up, if we're catching "valid" issues?

The one concern I have is that it's likely we'll trigger behaviour that _only_ happens the first time a form (or any of the other request data tbh) is accessed. 

For example, take this case. We call `EnsureForm()` and that "fills" the form, which triggers an exception. What happens if the customer calls `request.Form` later? Is `EnsureForm` called again and the exception thrown again? If so, that's great. But if _not_ (because it's only called once), then we're potentially introducing hard-to-debug issues... 

I _think_ the behaviour is what really matters here. e.g. what happens if a _customer_ calls `request.Form` again in this scenario? Is that something we can test (or just confirm from the code)

Reply:
```
    internal HttpValueCollection EnsureForm()
    {
        if (_form == null)
        {
            _form = new HttpValueCollection();
            if (_wr != null)
            {
                FillInFormCollection();
            }

            _form.MakeReadOnly();
        }

        return _form;
    }
```

Good point. FillInFormCollection() can potentially throw an HttpException (confirmed by looking at the code) and the next time, _form would not be null anymore so, yes, we might encounter situations where we could be modifying the result obtained by the customers code.

This is something that will happen anyway. If we catch or not the exception will not change that. On the other hand, we really need to read the body of the request. I wonder if there is any possible way that we could read the body of the request without causing this potential behavior change.

About catching exceptions, I tried to be restrictive here, catching only the exceptions types that we have reviewed. If a new exception type is thrown, it would probably be worth to log it and check if we should just catch it or if we should do anything else.